src/app/profile-edit/profile-edit.page.ts

```diff
@@ -176,6 +183,7 @@ export class ProfileEditPage implements isDeactivatable{
   onOptionChange(event: any) {
     const { event: selectedEvent, control } = event;
```

Comment:
in optionchange and optionchange2 fucntion some of the logic can be extracted into a single function check if its feasible to do so